R/zchunk_L141.hfc_R_S_T_Y.R

```diff
@@ -51,67 +48,210 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
+      left_join(L141.EDGAR_hfc_R_S_T_Yh.long,
+                by = c("EDGAR_agg_sector", "GCAM_region_ID", "year", "Non.CO2")) %>% # there should be NAs as some sectors have no emissions
+      replace_na(list(emissions=0)) # replace NAs with zero
+
+    # Disaggregate HFC cooling emissions to residential and commercial sectors
+    # Calculate share of res/com cooling emissions from L144.in_EJ_R_bld_serv_F_Yh
+    # Apply share to HFC emissions
+
+    # Select residential and cooling emisssions from L144.in_EJ_R_bld_serv_F_Yh
+    L141.R_cooling_T_Yh.long <- L144.in_EJ_R_bld_serv_F_Yh %>%
+      filter(service %in% c("comm cooling", "resid cooling"), fuel == "electricity")
+    # Group by GCAM region and ID and year in new data frame (use to calculate share of total later)
+    L141.R_cooling_Yh <-  L141.R_cooling_T_Yh.long %>%  group_by(GCAM_region_ID,year) %>%
+      summarize(total = sum(value))
+    # join in the totals we just calculated and calulate the shares
+    L141.R_cooling_T_Yh.long <- left_join_error_no_match(L141.R_cooling_T_Yh.long,L141.R_cooling_Yh, by = c("GCAM_region_ID", "year")) %>%
+      mutate(share = value / total) %>%
+      select(GCAM_region_ID,year,service,share,value)
+
+    # Add res/com cooling share to HFC and make adjustment
+    L141.hfc_R_S_T_Yh_share<- left_join(L141.hfc_R_S_T_Yh.long,
+                                        L141.R_cooling_T_Yh.long ,
+                                        by = c("GCAM_region_ID", "year", "supplysector" = "service")) %>% # there should be NAs.
+                                        #there are sectors that don't have emissions
+      replace_na(list(share=1)) %>% # replace those shares with "1"
+      mutate(emissions = emissions * share)
+
+    # Scale HFC Emissions to SSP2 HFC (Guus Velders) inventory (year 2010), where available
+
+    # EDGAR data doesn't have 2010 data, so match EDGAR 2008 data with GV 2010 data
+    # Duplicate EDGAR 2008 data and rename as 2010
+    TEMP <- subset(L141.hfc_R_S_T_Yh_share, year == 2008)
+    TEMP$year <- 2010
+    L141.hfc_R_S_T_Yh_share <- bind_rows(L141.hfc_R_S_T_Yh_share, TEMP)
+
+    # Process SSP2 HFC (Guus Velders) inventory
+    HFC_Inventory_GV <- HFC_Inventory_GV %>% mutate(Species = gsub("-","", Species)) %>%
+      mutate(Species = gsub("4310mee", "43",Species)) %>%
+      rename(SSP2_tot = Emissions)
+
+    # Add SSP2 values to EDGAR HFC emissions then calculate scalar
+    L141.hfc_scaler <- L141.hfc_R_S_T_Yh_share %>% group_by(year,Non.CO2) %>%
+      summarize(EDGAR_tot = sum(emissions))%>%
+      arrange(Non.CO2,year) %>%
+      left_join(HFC_Inventory_GV, by = c(year = "Year", Non.CO2 = "Species")) %>% # some entries not in GUUS data, default to scaler =1
+      mutate(scaler = SSP2_tot / EDGAR_tot) %>%
+      replace_na(list(scaler = 1)) %>%
+      mutate_at(vars(), funs(replace(., is.infinite(.), 1))) %>% # replace na and Infinite values with 1.
```

Comment:
Why? This step isn't in the original code that I see (lines 121-123 in `L141.hfc_R_S_T_Y.R`).